File: Data Structures & Algorithms/top-k-elements-in-list/submission-0.py
class Solution:
    def topKFrequent(self, nums: List[int], k: int) -> List[int]:
        counter = []

        for i in range(20001):
            counter.append(0)
    
        for i in range(len(nums)):
            if (nums[i] < 0):
                counter[abs(nums[i]) + 10000] += 1
            else:
                counter[nums[i]] += 1

        res = []
        for i in range(k):
            max_index = counter.index(max(counter))

            if (max_index <= 10000):
                res.append(max_index)
                counter[max_index] = 0
            else:
                x = max_index - 10000
                res.append(-1*x)
                counter[max_index] = 0

        return res

File: Data Structures & Algorithms/maximum-odd-binary-number/submission-0.cpp
class Solution {
public:
    string maximumOddBinaryNumber(string s) {
        int n = s.size(); int count = 0;

        for (char& c: s) {
            if (c == '1') {
                count++;
            }
        }
        vector<int> res(n, 0);
        if (count == 1) {
            res[n-1] = 1;
        }
        else {
            res[n-1] = 1;
            count--;
            int index = 0;
            while (count) {
                res[index] = 1;
                count--;
                index++;
            }
        }

        string temp = "";

        for (int& e: res) {
            temp.push_back(char(e + '0'));
        }

        return temp;
        }
};